SocialCareCaseViewerApi.Tests/V1/Controllers/FormSubmissionControllerTests.cs

```diff
@@ -256,5 +256,76 @@ public void GetSubmissionByQueryParametersReturns400WhenQueryCaseSubmissionsExce
             response?.StatusCode.Should().Be(400);
             response?.Value.Should().Be(errorMessage);
         }
+
+        [Test]
+        public void DeleteSubmissionReturns204WhenASubmissionIsSuccessfullyDeleted()
+        {
+            var request = TestHelpers.DeleteCaseSubmissionRequest();
+            var deletedSubmission = TestHelpers.CreateCaseSubmission();
+            _submissionsUseCaseMock.Setup(x => x.ExecuteDelete(deletedSubmission.SubmissionId.ToString(), request)).Verifiable();
+
+            var response = _formSubmissionController.DeleteSubmission(deletedSubmission.SubmissionId.ToString(), request) as NoContentResult;
```

Comment:
For all of these tests we need to do as ObjectResult.

The reason is I think the StatusCode gets cast when you specify the type of ObjectResult.

For instance if you change this to NotFoundObjectResult I expect the test would fail and start saying the StatusCode is 404.

As ObjectResult allows us to access the data without any casting occuring.

Reply:
This is interesting because if `ObjectResult` is used with `NoContent()` or `StatusCode(204)` the response is always null. I can't see how the controller setup could be changed to make that `as ObjectResult` work in the test.

I agree ObjectResult should be used, but looks like 204 is a special case that works a bit differently from other codes we normally use. 

I've changed one of the tests to use ObjectResult that should've had it.

As for your comment about changing the above to `NotFoundObjectResult`, that results in null response as well. It seems the conversion fails with anything other than `NoContentResult`

Happy to tweak this too if there's a way to handle it differently.